src/image/PhotoCalib.cc

```diff
@@ -397,6 +397,8 @@ void PhotoCalib::instFluxToMagnitudeArray(afw::table::SourceCatalog const &sourc
 std::shared_ptr<PhotoCalib> makePhotoCalib(daf::base::PropertySet &metadata, bool strip) {
     double calibration = 0.0, calibrationErr = 0.0;
 
+    // NOTE: we don't use makePhotoCalib(fluxMag0, fluxMag0Err) here because 0 were the "make a default"
+    // Calib values, and we don't want that to result in non-finite PhotoCalibs.
```

Comment:
I don't understand this comment. What defaults? I don't see any special handling of 0 appearing in the metadata.

Reply:
`Calib` only had a default constructor, which set those to (0,0). This meant that there are plenty of default-constructed `Calib`s (including in the metadata). The difference between the metadata version and this one is that I assume that if the metadata has 0, it means "construct the default version of the thing", whereas for this version I wanted to make it behave exactly as if you'd constructed a (0,0) `Calib` (which is a nonsensical object, but I'm "giving the user what they ask for").

Reply:
I'm not sure how to reword that though.

Reply:
I still don't understand. If `metadata.exists("FLUXMAG0")`, then the code you have is *exactly* equivalent to calling `makePhotoCalib(fluxMag0, fluxMag0Err)`. If not, then it returns `nullptr` (though this is not at all obvious; until I wrote this comment I thought it returns `PhotoCalib(0.0, 0.0)`).

It seems to me that if you called `makePhotoCalib`, you'd not only simplify the code inside the `if` statement, but you could make the execution path easier to understand as well.

Reply:
Ha! Ok, it looks like what I'd intended to write for the metadata version (what my comment above is trying to explain) was not what I actually implemented. So, yes, you're correct, and I'll just use this new function in the metadata one.

The nullptr-return for the metadata version is documented in its docstring. Relatedly, the name and docstring for `testCreateFromMetadataRaisesNoKey` need to be changed: I'd originally planned on raising for that case, but was convinced to return nullptr instead (for backwards compatibility with old files that don't have Calib info in them).